sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java

```diff
@@ -72,9 +78,31 @@ public void execute(SensorContext context) {
     List<InputFile> list = new ArrayList<>();
     it.forEach(list::add);
     List<InputFile> inputFiles = Collections.unmodifiableList(list);
-
+    Map<URI, String> inputFileByPackage = getInputFileByPackage(context.fileSystem().baseDir(), inputFiles);
     PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, inputFiles);
-    scanner.scanFiles();
+    scanner.scanFiles(inputFileByPackage);
+  }
+  // visible for testing
+  static Map<URI, String> getInputFileByPackage(File projectBaseDir, List<InputFile> inputFiles) {
+    Map<URI, String> packageNamePerFile = new HashMap<>();
+    for (InputFile inputFile : inputFiles) {
+      String packageName = findInitFiles(inputFile, projectBaseDir);
+      packageNamePerFile.put(inputFile.uri(), packageName);
+    }
+    return packageNamePerFile;
+  }
```

Comment:
Do we really need this method? Do we need to build a map for all InputFiles? What about moving `findInitFiles` to `PythonScanner` and call it inside `scanFile`?

Reply:
Good point, I will do that